### PR TITLE
Scenario Result

### PR DESCRIPTION
**Pull request by hugoiman**
PR: change return []TestCase to ScenarioResult in func Run()

### CODE REVIEW

File: scenario/scenario.go [hunk 1]
```diff
@@ -280,5 +280,8 @@ loop:
 		}
 	}
 
-	return testResults
+	sr := restify.ScenarioResult(testResults)
+	sr.IsFailed()
```
Comment (bastianrob) [inline scenario/scenario.go]: kenapa manggil `.IsFailed()` disini??

**Review by bastianrob — CHANGES_REQUESTED** [review]

File: scenario/scenario.go (hunk 1 above)
Comment (hugoiman) [reply scenario/scenario.go]: oiyak lupa di apus

File: scenario/scenario.go [hunk 2]
```diff
@@ -280,5 +280,7 @@ loop:
 		}
 	}
 
-	return testResults
+	sr := restify.ScenarioResult(testResults)
```
Comment (bastianrob) [inline scenario/scenario.go]: Ini juga sebernya gak perlu di tampung ke `sr` langsung aja `return restify.ScenarioResult(testResults)`

**Review by bastianrob — CHANGES_REQUESTED** [review]

File: result.go [hunk 3]
```diff
@@ -40,3 +40,17 @@ func NewTestResult(scn Scenario, tcn int) TestResult {
...
 	}
 }
+
+//ScenarioResult is
+type ScenarioResult []TestResult
+
+//IsFailed is
+func (sr *ScenarioResult) IsFailed() bool {
+	for _, tc := range *sr {
+		if tc.Success == false {
```
Comment (bastianrob) [inline result.go]: `if !tc.Success`

**Review by bastianrob — CHANGES_REQUESTED** [review]

File: result.go [hunk 4]
```diff
@@ -40,3 +40,17 @@ func NewTestResult(scn Scenario, tcn int) TestResult {
 		ExpectedCode:  tc.Expect.StatusCode,
 	}
 }
+
+//ScenarioResult is
+type ScenarioResult []TestResult
+
+//IsFailed is
```
Comment (bastianrob) [inline result.go]: IsFailed is what?
Please give a proper comment

File: result.go [hunk 5]
```diff
@@ -40,3 +40,17 @@ func NewTestResult(scn Scenario, tcn int) TestResult {
 		ExpectedCode:  tc.Expect.StatusCode,
 	}
 }
+
+//ScenarioResult is
```
Comment (bastianrob) [inline result.go]: ScenarioResult is what?
Please give a proper comment

**Review by bastianrob — CHANGES_REQUESTED** [review]

**Comment by bastianrob** [comment]
@hugoiman

**Comment by bastianrob** [comment]
Anyway ini juga kenapa masih di branch `feature/time-delay` ya

**Review by bastianrob — APPROVED** [review]

**Pull request merged**